frontend_react/src/components/Map/Map.tsx

```diff
@@ -61,10 +78,14 @@ const Map = (props: MapProps) => {
   // Update the markers on the map when historical or live users are added
   useEffect(() => {
     if (!_.isNil(groupedUsers)) {
+      let locationList: any = [];
       const markers: ReactElement[] = [];
       Object.entries(groupedUsers).forEach(([, users], index) => {
```

Comment:
Seems like this UseEffect is called twice whenever the map is loaded. Maybe because groupedUsers and historicalUsers both change at the same time? Maybe @arsalanfardi can shine a light on this. But this causes the map to slow down for 2-3 seconds and show a shit ton of markers before adjusting and reducing to 100

Reply:
You could try removing historicalUsers from the dependency of the useEffect to see if it helps. It's not actually required since like you noticed.. React just complains if you don't include every variable inside the useEffect dependencies